### PR TITLE
Compare: shared brush selection + color-by-column, cached distance backend (#131, #132)

### PR DESCRIPTION
**Pull request by enjalot**
Makes the Compare page a real projection-comparison tool and turns its distance code into a shared, cached resource. Closes #131 and #132.

## Backend — distance is now a cached, shared resource (`search.py`)
The Compare endpoints re-read both umap parquets and **re-fit a full `NearestNeighbors` index on every request** — and `/compare/neighbors` did it on *every click*.
- Added `_umap_coords` / `_umap_knn` (`UMAP_COORDS` / `UMAP_KNN` LRUs, mirroring the existing `NN_CACHE` pattern). Coordinates and fitted kNN are built once and reused.
- Vectorized the `relative` metric (dropped its Python per-point loop); `neighborhood` reuses cached neighbor matrices.
- `/compare/neighbors` now reads the cached neighbor matrix instead of refitting → clicks are effectively instant.

## #131 — continuous color-by column (Compare)
- New `GET /api/datasets/<ds>/column/<col>` — numeric column values + extent from the cached `input.parquet` (400 on non-numeric).
- "Color by" picker in `CompareControls` (Drift metric or any numeric column), normalized into the existing continuous color channel, with a new `ColorLegend` ramp. Threshold slider hides in column mode.

## #132 — shared brush selection + spread stat
- `Scatter.jsx` distinguishes a Shift-drag lasso from a single click (via `lassoEnd`→`select` ordering). This also **fixes a latent bug**: a Shift-lasso previously mis-fired a single-point neighbor fetch.
- Brushing in either pane highlights the same rows **green in both panes** via a new `SelectionOverlay` (the existing `NeighborPlot`/`CrosshairPlot` overlay pattern) — chosen because regl's `pointColorActive` is silently ignored under `colorBy:'valueB'`.
- New `POST /api/search/compare/spread` + a spread readout in `CompareDataPanel`: mean pairwise distance + convex-hull area per side with the R/L ratio ("2.3× more spread out in the right map"). O(n²) mean-pairwise is subsampled above 2000 points.

The two features compose: brush a suspicious region while colored by an external error column.

## Tests / checks
- 13 new backend tests (`tests/test_compare.py`): metrics, cached neighbors, column endpoint (+400 on text), spread stats + edge cases. Full suite: no new failures.
- `ruff check` clean; frontend `eslint` clean, `vitest` 82 pass, production build succeeds.
- Verified end-to-end over real HTTP against a built 2-umap dataset (all metrics, neighbors, column, spread).

> Note: base is `sprite-atlas` (the Compare page lives there, not yet on `main`).
> Manual browser QA of the Shift+drag lasso is still recommended — it's a WebGL interaction unit tests can't exercise.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: latentscope/server/app.py [hunk 1]
```diff
@@ -330,6 +330,54 @@ def query():
...
+            extent = None
+        if not extent or extent[0] is None or extent[1] is None:
+            valid = series.replace([np.inf, -np.inf], np.nan).dropna()
+            if len(valid):
+                extent = [float(valid.min()), float(valid.max())]
+            else:
+                extent = [None, None]
+
+        # NaN/inf aren't valid JSON; null them so the client can skip them.
+        values = series.to_numpy(dtype="float64")
```
[the remark this answers is not in the document]
Comment (enjalot) [reply latentscope/server/app.py]: Fixed in 59ceadd — pass `na_value=np.nan` to `to_numpy(dtype="float64")` so nullable (Int64/Float64) and Arrow-backed numeric columns with missing values coerce NA→NaN (then serialize as `null`) instead of raising. Added a test with a nullable `Int64` column containing `pd.NA`.
